tests/integration/stepfunctions/conftest.py

```diff
@@ -0,0 +1,213 @@
+                "Principal": {"Service": "states.amazonaws.com"},
+                "Action": "sts:AssumeRole",
+            }
+        ],
+    }
+    snf_permission = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "lambda:*",
+                    "sqs:*",
+                    "dynamodb:*",
+                    "secretsmanager:*",
+                    "logs:*",
+                ],
+                "Resource": ["*"],
+            }
+        ],
+    }
+
+    def _create():
+        return create_iam_role_with_policy(
+            RoleName=role_name,
+            PolicyName=policy_name,
+            RoleDefinition=snf_role,
+            PolicyDefinition=snf_permission,
+        )
+
+    return _create
+
+
+_state_machine_arns: Final[list[str]] = list()
+
+
+@pytest.fixture
+def create_state_machine(stepfunctions_client):
+    def _create_state_machine(**kwargs):
+        create_output = stepfunctions_client.create_state_machine(**kwargs)
+        create_output_arn = create_output["stateMachineArn"]
+        _state_machine_arns.append(create_output_arn)
+        return create_output
+
+    yield _create_state_machine
+
+    for state_machine_arn in _state_machine_arns:
+        try:
+            stepfunctions_client.delete_state_machine(stateMachineArn=state_machine_arn)
+        except Exception:
+            LOG.debug(f"Unable to delete state machine '{state_machine_arn}' during cleanup.")
+
+
+@pytest.fixture
+def await_no_state_machines_listed(stepfunctions_client):
+    def _exec_await():
+        def _is_empty_state_machine_list():
+            lst_resp = stepfunctions_client.list_state_machines()
+            state_machines = lst_resp["stateMachines"]
+            return not bool(state_machines)
+
+        success = poll_condition(
+            condition=_is_empty_state_machine_list,
+            timeout=_DELETION_TIMEOUT_SECS,
+            interval=5,
+        )
+        if not success:
+            LOG.warning("Timed out whilst awaiting for listing to be empty.")
+
+    return _exec_await
+
+
+def _is_state_machine_listed(stepfunctions_client, state_machine_arn: str) -> bool:
+    lst_resp = stepfunctions_client.list_state_machines()
+    state_machines = lst_resp["stateMachines"]
+    for state_machine in state_machines:
+        if state_machine["stateMachineArn"] == state_machine_arn:
+            return True
+    return False
+
+
+@pytest.fixture
+def await_state_machine_not_listed(stepfunctions_client):
+    def _exec_await(state_machine_arn: str):
+        success = poll_condition(
+            condition=lambda: not _is_state_machine_listed(stepfunctions_client, state_machine_arn),
+            timeout=_DELETION_TIMEOUT_SECS,
+            interval=5,
+        )
+        if not success:
+            LOG.warning(f"Timed out whilst awaiting for listing to exclude '{state_machine_arn}'.")
+
+    return _exec_await
```

Comment:
As already discussed, I'd suggest not making this a fixture. Just clutters the already quite cluttered fixtures, doesn't use a lifecycle and will be easier to use with different client configurations.